src/main/java/longah/LongAh.java

```diff
@@ -1,21 +1,173 @@
+                    System.out.println(otherPerson + " owes " + name + " $" + formattedAmount);
+                    hasDebts = true;
+                } else if (amount < 0 && balances.containsKey(otherPerson) && balances.get(otherPerson) == -amount) {
+                    // Skip, as the other person already owes this person
+                } else {
+                    System.out.println(name + " owes " + otherPerson + " $" + formattedAmount);
+                    hasDebts = true;
+                }
+            }
+        }
+
+        if (!hasDebts) {
+            System.out.println("No debts, Huat Ah!");
+        }
+    }
+
+    /**
+     * Gets the other person involved in a transaction.
+     *
+     * @param name The name of the person.
+     * @return The name of the other person.
+     */
+    private String getOtherPerson(String name) {
+        return transactions.getOtherPerson(name);
+    }
+
+    /**
+     * Deletes a debt transaction by index.
+     *
+     * @param index The index of the transaction to delete.
+     */
+    public void deleteDebt(int index) {
+        transactions.remove(index);
+    }
+
+    /**
+     * Finds debts involving a specific person.
+     *
+     * @param person The name of the person to find debts for.
+     */
+    public void findDebts(String person) {
+        for (Transaction transaction : transactions.getTransactions()) {
+            if (transaction.getInvolves(person)) {
+                System.out.println(transaction);
+            }
+        }
+    }
+
+    /**
+     * Clears all debts.
+     */
+    public void clearAllDebts() {
+        transactions.clear();
+    }
+
+    /**
+     * Gets or creates a member with the given name.
+     *
+     * @param name The name of the member.
+     * @return The Member object.
+     */
+    private Member getOrCreateMember(String name) {
+        return members.computeIfAbsent(name, Member::new);
+    }
+
     /**
-     * Main entry-point for the java.duke.Duke application.
+     * The main method to run the LongAh application.
+     *
+     * @param args The command-line arguments.
      */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-        System.out.println("What is your name?");
-
-        Scanner in = new Scanner(System.in);
-        System.out.println("Hello " + in.nextLine());
+        System.out.println("Welcome to LongAh!");
+        LongAh app = new LongAh();
+        while (true) {
+            System.out.println("Enter command:");
+            if (!app.scanner.hasNextLine()) {
+                return;
+            }
+            String command = app.scanner.nextLine();
+            String[] parts = command.split(" ");
```

Comment:
Blank space splitting may not be the best idea, especially if names have blank space within them.

Reply:
Split by `p/` followed by a further split by `a/` might be better, throwing exceptions if `a/` does not exist to signify invalid inputs unless we decide to add functionality for that.